localstack/services/dynamodb/server.py

```diff
@@ -108,6 +109,15 @@ def create_dynamodb_server(port=None) -> DynamodbServer:
         mkdir(ddb_data_dir)
         absolute_path = os.path.abspath(ddb_data_dir)
         server.db_path = absolute_path
+    else:
```

Comment:
I'd suggest that we slightly change the implementation here, to make it easier to patch it elsewhere, as needed:

Something like:
```
def create_dynamodb_server(port: int = None) -> DynamodbServer:
    """
    Creates a dynamodb server from the LocalStack configuration.
    """
    port = port or get_free_tcp_port()
    ddb_data_dir = f"{config.dirs.data}/dynamodb" if config.dirs.data else None
    return do_create_dynamodb_server(port, ddb_data_dir)


def do_create_dynamodb_server(port: int, ddb_data_dir: str) -> DynamodbServer:
    if ddb_data_dir:
        mkdir(ddb_data_dir)
        absolute_path = os.path.abspath(ddb_data_dir)
        server.db_path = absolute_path

    server.heap_size = config.DYNAMODB_HEAP_SIZE
    ...
```

Reply:
@whummer nice suggestion 😄 ! It's  indeed way more flexible this way. 
Have just two tiny questions about it. 
- How/where do you want to patch `do_create_dynamodb_server` such that we pass a temporary directory (`config.dirs.tmp`) as `ddb_data_dir`?
- Do we delete temporary directories when we start/stop LocalStack. If not, shouldn't we also make sure to remove the temporary directory when starting the server? Otherwise, such a temporary directory would act basically like as a persistent folder (`DATA_DIR`).

Reply:
Nice suggestions 😄 patching `do_create_dynamodb_server` would give us much more flexibility.
Adapted the PR accordingly.